src/packagedcode/pypi.py

```diff
@@ -390,7 +390,11 @@ def assign_package_to_resources(cls, package, resource, codebase, package_adder)
         Assign files to package for an installed wheel. This requires a bit
         of navigation around as the files can be in multiple places.
         """
-        site_packages = resource.parent(codebase).parent(codebase)
+        dist_info_dir = resource.parent(codebase)
+        if not dist_info_dir:
+            return
+
```

Comment:
Let also check that this is a proper dist-info dir and that this match the name pattern.
```suggestion
        if not dist_info_dir.name.endswith(".dist-info"):
            return
```

Reply:
This is not required as we have the following as path_patterns: `path_patterns = ('*.dist-info/METADATA',)` and so only resources matching this path pattern will have this function invoked.